pkg/controllers/subnetbinding/subnetbinding_controller.go

```diff
@@ -270,15 +326,82 @@ func (r *Reconciler) validateVpcSubnetsBySubnetCR(ctx context.Context, namespace
 		bms := r.SubnetBindingService.GetSubnetConnectionBindingMapsByChildSubnet(subnets[0])
 		if len(bms) > 0 {
 			dependency := r.SubnetBindingService.GetSubnetConnectionBindingMapCRName(bms[0])
-			return nil, &errorWithRetry{
+			return nil, subnetCR, &errorWithRetry{
 				message: fmt.Sprintf("Target Subnet CR %s is associated by %s", name, dependency),
 				error:   fmt.Errorf("target Subnet %s is already associated by SubnetConnectionBindingMap %s", name, dependency),
 				retry:   true,
 			}
 		}
 	}
+	return subnets, subnetCR, nil
+}
 
-	return subnets, nil
+func (r *Reconciler) getSharedSubnetFromNSX(subnet *v1alpha1.Subnet) (*model.VpcSubnet, *errorWithRetry) {
```

Comment:
Is it better if we move this function to SubnetService? Even more, we can let the SubnetService to check if the Subnet CR is shared or not, then decide how to list the NSX VpcSubnets, e.g., from local cache or from NSX API request. I guess this logic is also callable by SubnetController when reconciling (including deleting) the CR ?